src/config.ts

```diff
@@ -148,7 +149,8 @@ class ConfigurationFactory {
         const url = githubEvent["repository"]
           ? githubEvent["repository"]["url"]
           : `${process.env.GITHUB_SERVER_URL}/${process.env.GITHUB_REPOSITORY}`
-        return `${url}/tarball/${ref}`
+        const tarballUrl = `${url}/tarball/${ref}`
+        return tarballUrl.includes("#") ? tarballUrl.replace(/#/g, '%23') :tarballUrl
```

Comment:
@gdelgadot  shouldn't we just `urlencode` the whole branch ref preventing any other character issues, instead doing a single replace? 

Theoretically, all after `tarball/` would be parsed like a single variable, so it would be important to `urlencode` all its content. Otherwise, we could have the same error with any other character that break the format (maybe `:`, `\`... etc).

Reply:
got it